### DISCUSSION TITLE
Avoid setting a FIDO2 PIN when using a hardware key for GitHub's 2FA

### DISCUSSION THREAD

**Discussion by hackerb9** [community/community#64672] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Dear GitHub,

Please change GitHub's 2FA implementation so that the `discoverableCredential` flag on the WebAuthn call is set to **DISCOURAGED**, which is the correct setting in a two-factor scenario.  Currently GitHub appears to be leaving discoverableCredential unspecified, which most browsers will treat the same as PREFERRED, meaning a device without a PIN cannot register with GitHub. Instead, the user will see an error, such as, 

> User verification failed on github.com. You may need to set a PIN on your device. <kbd>Cancel</kbd>

When I registered my [Solokeys](http://solokeys.com) for two-factor authentication today, I did not want to set a pin. The workaround that I used was setting `security.webauthn.ctap2` to **False** in Firefox's Advanced Settings ([about:config](about:config)). 

That worked for the moment, but it would be better if GitHub fixed this issue for everyone. Sooner would be better than later, as GitHub is rolling out 2FA to everyone and, once a person adds a PIN to a key, they will be asked for it every time they register anywhere. 

Thank you.

_____

<sub><i>

Edit: I have been informed that the userVerification flag is not the problem, that instead the issue is the discoverableCredential, and modified the above text accordingly.

</i></sub>

**Comment by hackerb9** [comment]
My apologies, this bug has been resolved.

**Reply by tmitchell** [reply, marked as answer]
I wonder if there has been a regression on this issue?  I currently have a FIDO2 hardware key (without a PIN) joined from earlier in the year that works for GitHub authentication, but joining a new one today with Firefox I'm encountering the error message in question:
> User verification failed on github.com. You may need to set a PIN on your device. `Cancel`

Update: Did some more digging when I saw this happen on other sites besides GitHub.  Appears to be a change in Firefox that caused this.  Confirmed fix with Firefox Beta which includes this: https://bugzilla.mozilla.org/show_bug.cgi?id=1530370.
